### PR TITLE
Thermomachines no longer self-destruct when built on blocked ports

### PR DESCRIPTION
**Pull request by RikuTheKiller**
## About The Pull Request

Ever felt the utter pain when you make a new thermomachine, hook it up perfectly and then screwdriver it only for it to immediately become a pile of iron and components without warning? This PR fixes that.

Instead of doing what it did previously, it unanchors itself and opens its panel.

Right, almost forgot to mention, failing to wrench down a thermomachine no longer bonks it with the wrench.

Due to not being allowed to use visual messages for when the port is blocked, I've added a mob/user variable to all on_construction() procs. (This allowed me to use balloon messages instead.)
## Why It's Good For The Game

Saves a lot of unnecessary headaches when working with thermomachines.
## Changelog
:cl:
fix: Atmosians have finally convinced the thermomachines to not self-destruct when built on blocked ports.
fix: Failing to wrench down a thermomachine no longer hits it with the wrench.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
We are in a feature freeze which includes QoL changes (since what defines one is far less objective than bug fix), please reopen after the freeze

**Comment by Mothblocks** [comment]
Wait this isn't qol at all why did you tag it that

**Comment by RikuTheKiller** [comment]
Deconstructing was intended behaviour, that's why I tagged it as such. I'm plenty okay with making it a fix, though!

File: code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm [hunk 1]
```diff
@@ -55,10 +55,14 @@
...
+
 	if(check_pipe_on_turf())
-		deconstruct(TRUE)
-		return
-	return..()
+		set_anchored(FALSE)
+		set_panel_open(TRUE)
+		change_pipe_connection(TRUE)
+		icon_state = "thermo-open"
+		visible_message(span_warning("You fail to anchor the thermomachine as it's port is being hogged up by something else."))
```
Comment (Mothblocks) [inline code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm:65]: this should be a balloon alert, right now not only is it a pointless chat, but it's also being sent to everyone as "you"

**Comment by Mothblocks** [comment]
maybe ill confused so ill just do nothing

File: code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm (hunk 1 above)
Comment (RikuTheKiller) [reply code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm:65]: Should I change the other visible message as well?

Comment (RikuTheKiller) [reply code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm]: I've made it no longer address people as "you", but I'm also not sure how I'm meant to turn it into a balloon alert without a mob to send it to. It's also a pretty long message and I have no idea on how to convey it other than just "Anchor failed" which sounds like shit.

**Comment by RikuTheKiller** [comment]
> maybe im confused so ill just do nothing

confused moth

File: code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm [hunk 2]
```diff
@@ -55,10 +55,14 @@
...
+
 	if(check_pipe_on_turf())
-		deconstruct(TRUE)
-		return
-	return..()
+		set_anchored(FALSE)
+		set_panel_open(TRUE)
+		change_pipe_connection(TRUE)
+		icon_state = "thermo-open"
+		visible_message(span_warning("The thermomachine is left unanchored as it's port is being hogged up by something else."))
```
Comment (Mothblocks) [inline code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm:65]:
```suggestion
		balloon_alert(/* figure out the user here, visible_message should not be used as a hack */, "the port is already being used!")
```

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Balloon is necessary

File: code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm (hunk 2 above)
Comment (RikuTheKiller) [reply code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm:65]: It's now a balloon alert...

Comment (Mothblocks) [reply code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm:65]: ?? No it is not this, is the latest commit?

Comment (RikuTheKiller) [reply code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm:65]: Apparently I may have forgotten to sync it?

**Pull request merged**